Models/Developer.cs

```diff
@@ -0,0 +1,13 @@
+﻿
+namespace EmployeesSalaries.Models
+{
+    public class Developer: Employee, ISubordinate<LeadDeveloper>
```

Comment:
great idea to add an interface, but the way you have done it i'm not sure if it will work
replace it with an interface called IManaged for example, and no need for it to be generic for now